zipline/pipeline/loaders/events.py

```diff
@@ -0,0 +1,154 @@
+                            "Pass `infer_timestamps=True` to use the first "
+                            "date in `all_dates` as implicit timestamp." %
+                            (TS_FIELD_NAME, k)
+                        )
+                    self.events_by_sid[k] = v = v.copy()
+                    v.index = [dates[0]] * len(v)
+                else:
+                    self.events_by_sid[k] = v.set_index(TS_FIELD_NAME)
+            else:
+                raise ValueError("Data for sid %s must be in DataFrame, "
+                                 "Series, or DatetimeIndex." % k)
+            # Once data is in a DF, make sure columns are correct.
+            cols_except_ts = (set(v.columns.values) -
+                              {TS_FIELD_NAME} -
+                              {SID_FIELD_NAME})
+            # Check that all columns other than timestamp are as expected.
+            if cols_except_ts != expected_cols:
+                raise ValueError(
+                    "Expected columns %s for sid %s but got columns %s." %
+                    (expected_cols, k, v.columns.values)
+                )
+        self.dataset = dataset
+
+    def get_loader(self, column):
+        if column in self.dataset.columns:
+            return getattr(self, "%s_loader" % column.name)
+        raise ValueError("Don't know how to load column '%s'." % column)
+
+    def load_adjusted_array(self, columns, dates, assets, mask):
+        return merge(
+            self.get_loader(column).load_adjusted_array(
+                [column], dates, assets, mask
+            )
+            for column in columns
+        )
+
+    def mk_date_series(self, date_field_name):
+        return {sid: pd.Series(index=event.index,
+                               data=np.array(event[date_field_name]))
+                for sid, event in iteritems(self.events_by_sid)}
+
+    def _next_event_date_loader(self, next_date_field, event_date_field_name):
+        return DataFrameLoader(
+            next_date_field,
+            next_date_frame(
+                self.all_dates,
+                self.mk_date_series(event_date_field_name),
+            ),
+            adjustments=None,
+        )
+
+    def _previous_event_date_loader(self,
+                                    prev_date_field,
+                                    event_date_field_name):
+        return DataFrameLoader(
+            prev_date_field,
+            previous_date_frame(
+                self.all_dates,
+                self.mk_date_series(event_date_field_name),
+            ),
+            adjustments=None,
+        )
+
+    def _previous_event_value_loader(self,
+                                     previous_value_field,
+                                     event_date_field_name,
+                                     value_field_name):
+        return DataFrameLoader(
+            previous_value_field,
+            previous_value(
```

Comment:
There is now asymmetry between `previous_date_frame` and `previous_value_field`. How do you feel about making `previous_date_field` accept the column names and do the slicing internally to be more consistent. The old API made more sense when we were storing the data as series but this is no longer true.